nemo_text_processing/inverse_text_normalization/hy/taggers/time.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pynini
+from nemo_text_processing.inverse_text_normalization.hy.utils import get_abs_path
+from nemo_text_processing.text_normalization.en.graph_utils import GraphFst, delete_space
+from pynini.lib import pynutil
+
+
+class TimeFst(GraphFst):
+    """
+    Finite state transducer for classifying time
+        e.g. տասներկուսն անց հինգ -> time { hours: "12" minutes: "05" }
+        e.g. հինգին տասնհինգ պակաս -> time { hours: "04" minutes: "45" }
+        e.g. տասներեք անց կես -> time { hours: "12" minutes: "30" }
+    """
+
+    def __init__(self):
+        super().__init__(name="time", kind="classify")
+        graph_oclock = pynini.accep("անց")
+        graph_oclock = pynutil.delete(graph_oclock)
+
+        graph_demi = pynini.accep("կես")
+        graph_demi = pynini.cross(graph_demi, "30")
+
+        graph_fractions = graph_demi
+
+        graph_hours = pynini.string_file(get_abs_path("data/time/hours.tsv")) + (
+            pynutil.delete("") | pynutil.delete("ն") | pynutil.delete("ին")
```

Comment:
As before, this is an empty graph so unsure of its purpose. If you want to simply make these unions optional, you can just use the closure property `pynini.closure(...,0,1)` which is more explicit. (There may also be a `.ques` shorthand but have to double check.

Reply:
Yeah, you can have the same logic with pynini.union(....).ques

Reply:
Thank you very much, as you can see I've changed that line everywhere you mentioned with pynini.closure.